### PR TITLE
b256 using wide ops

### PR DESCRIPTION
**Pull request by xunilrj**
## Description

This PR change `b256` bitwise operations implementation to use wide operators.
It does not add any new functionality to `b256`.

Other changes are related to `clippy` complaining about some clones.

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by IGI-111 — APPROVED** [review]

**Review by anton-trunov — COMMENTED** [review]
LGTM! Just a minor thing: let's add some const-eval tests for the `b256` intrinsics to `test/src/e2e_vm_tests/test_programs/should_pass/language/const_inits/src/main.sw`

**Review by anton-trunov — DISMISSED** [review]

**Review by JoshuaBatty — DISMISSED** [review]

**Review by IGI-111 — DISMISSED** [review]

**Review by anton-trunov — APPROVED** [review]

**Comment by xunilrj** [comment]
Following the explanation at https://github.com/FuelLabs/sway/issues/5121. My suggestion is to merge this PR with the register allocation changes. It solves the non-determinism and performance is on par with what we have previously.

File: sway-core/src/asm_generation/fuel/register_allocator.rs [hunk 1]
```diff
@@ -777,6 +777,7 @@ fn assign_registers(
             .neighbors_undirected(node)
             .map(|neighbor| interference_graph[neighbor].clone())
             .collect();
+        // println!("assign_registers: {:?}", neighbors);
```
Comment (anton-trunov) [inline sway-core/src/asm_generation/fuel/register_allocator.rs:780]: Do we want to keep this commented-out line?

**Review by anton-trunov — APPROVED** [review]

File: sway-core/src/asm_generation/fuel/register_allocator.rs (hunk 1 above)
Comment (xunilrj) [reply sway-core/src/asm_generation/fuel/register_allocator.rs]: Yeah, no we don't. Removed.

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**